src/components/subscriptions/licenses/LicenseManagementTable/index.jsx

```diff
@@ -0,0 +1,296 @@
+  } = useContext(SubscriptionDetailContext);
+
+  const [users, forceRefreshUsers] = useSubscriptionUsers({
+    currentPage,
+    searchQuery,
+    subscriptionUUID: subscription.uuid,
+    errors,
+    setErrors,
+    userStatusFilter,
+  });
+
+  // Filtering and pagination
+  const updateFilters = (filters) => {
+    if (filters.length < 1) {
+      setSearchQuery(null);
+      setUserStatusFilter(null);
+    } else {
+      filters.forEach((filter) => {
+        switch (filter.id) {
+          case 'statusBadge':
+            setUserStatusFilter(filter.value.join(','));
+            break;
+          case 'email':
+            setSearchQuery(filter.value);
+            break;
+          default: break;
+        }
+      });
+    }
+  };
+
+  const debouncedUpdateFilters = debounce(
+    updateFilters,
+    DEBOUNCE_TIME_MILLIS,
+  );
+
+  const debouncedSetCurrentPage = debounce(
+    setCurrentPage,
+    DEBOUNCE_TIME_MILLIS,
+  );
+  // Call back function, handles filters and page changes
+  const fetchData = useCallback(
+    (args) => {
+      // pages index from 1 in backend, Datable component index from 0
+      if (args.pageIndex !== currentPage - 1) {
+        debouncedSetCurrentPage(args.pageIndex + 1);
+      }
+      debouncedUpdateFilters(args.filters);
+    },
+    [currentPage],
+  );
+
+  // Maps user to rows
+  const data = useMemo(
+    () => users?.results?.map(user => ({
+      id: user.uuid,
+      email: user.userEmail,
+      emailLabel: <span data-hj-suppress>{user.userEmail}</span>,
+      status: user.status,
+      statusBadge: <LicenseManagementUserBadge userStatus={user.status} />,
+      recent: userRecentAction(user),
+    })),
+    [users],
+  );
+
+  // Row action button functions
+  const rowRemindOnClick = (remindUser) => {
+    setRemindModal({
+      open: true,
+      users: [remindUser],
+    });
+  };
+  const rowRevokeOnClick = (revokeUser) => {
+    setRevokeModal({
+      open: true,
+      users: [revokeUser],
+    });
+  };
+
+  // Successful action modal callback
+  const onRemindSuccess = () => {
+    forceRefreshUsers();
+    setRemindModal(modalZeroState);
+    addToast(`User${revokeModal.users.length > 1 ? 's' : ''} successfully reminded`);
+  };
+  const onRevokeSuccess = () => {
+    // refresh subscription and user data to get updated revoke count and revoked user list
+    forceRefreshSubscription();
+    forceRefreshUsers();
+    forceRefreshOverview();
+    setRevokeModal(modalZeroState);
+    addToast(`License${revokeModal.users.length > 1 ? 's' : ''} successfully revoked`);
+  };
+
+  // Bulk Action buttons
+  const bulkRemindOnClick = (usersToRemind) => {
+    setRemindModal({
+      open: true,
+      users: usersToRemind,
+    });
+  };
+  const bulkRevokeOnClick = (usersToRevoke) => {
+    setRevokeModal({
+      open: true,
+      users: usersToRevoke,
+    });
+  };
+
+  return (
+    <>
+      {(users.count < 1 && userStatusFilter === null) && <SubscriptionZeroStateMessage /> }
```

Comment:
It seems like the `SubscriptionZeroStateMessage` flashes above the table briefly while the initial table data is loading.